### DISCUSSION TITLE
Looking for guidance on how to enable Micronaut AOT for an existing Micronaut AWS Lambda project

### DISCUSSION THREAD

**Discussion by nedelva** [micronaut-projects/micronaut-core#10555] [Q&A] [answered]
I want to experiment a little with the micronaut-aot feature in my Micronaut AWS Lambda project.

I started by looking at [this blog entry](https://micronaut.io/2021/12/20/micronaut-aot-build-time-optimizations-for-micronaut-applications/), which led me to [Cédric's blog post(outdated)](https://medium.com/graalvm/introducing-micronaut-aot-build-time-optimizations-for-your-micronaut-applications-68b8f1302c5). I say "outdated" because in 2021 there was no support for building AOT-enabled projects with Maven.

Luckily the Micronaut Launcher can provide the necessary plumbing, so I went ahead and created a project in it with two explicitly selected features : aws-lambda and micronaut-aot. Then I copied what I needed into my project:

- the aot-jar.properties to the root dir;
- added two Maven properties in my POM:
```
<micronaut.aot.enabled>false</micronaut.aot.enabled>
<micronaut.aot.packageName>com.example.aot.generated</micronaut.aot.packageName>
```
- added a `configuration` section for micronaut-maven-plugin:
```
<configuration>
    <configFile>aot-${packaging}.properties</configFile>
</configuration>
```

The default build process is running as before, but when running `mvn package -Dmicronaut.aot.enabled=true` I am getting this error:

> [INFO] --- mn:4.4.0:aot-analysis (default-aot-analysis) @ contracts ---
[INFO] Running Micronaut AOT 2.2.0 aot-analysis
[INFO] Packaging project
[ERROR] java.lang.UnsupportedClassVersionError: org/codehaus/plexus/classworlds/launcher/Launcher : Unsupported major.minor version 52.0

It looks like aot-analysis is trying to incorporate some parts of Maven itself(?) (AFAIK Plexus Launcher is an internal Maven dependency / dependency of one of the Maven plugins) 
[I am using Java 17 + Maven 3.9.5 and Micronaut 4.3.3]
How do I fix this?

**Comment by alvarosanchez** [comment, marked as answer]
This looks like a bug in the Maven Plugin. Can you please report it there?

**Comment by nedelva** [comment]
Thanks @alvarosanchez for your quick return. I followed your suggestion and reported a bug here: [issue#1009](https://github.com/micronaut-projects/micronaut-maven-plugin/issues/1009). 

It is worth mentioning that the issue does NOT reproduce on a MacBook Pro machine.
